lib/pharos/config.rb

```diff
@@ -140,5 +140,19 @@ def set(key, value)
     def to_yaml
       YAML.dump(to_h.deep_stringify_keys)
     end
+
+    # @example deep get network provider
+    #   config.deep_get("network.provider")
+    # @param key [String] a dot separated key, such as network.provider
+    # @return [Object,NilClass] nil when any part of the chain is unreachable
+    def deep_get(key)
```

Comment:
Probably should use `Pharos::CoreExt::DeepKeys`?

Reply:
This one doesn't need to hashify in the middle. Interestingly, something like `config.deep_get("hosts.first.address")` works with this.

Reply:
Is  one `.to_h` that bad? It would remove duplicate code.

Reply:
I threw away the DeepKeys refinement thing and replaced it with HashDiff and renamed `config.deep_get` to `config.dig` which acts like `hash.dig`: `config.dig(:network, :provider)`.